apricot/local/service.go

```diff
@@ -242,6 +242,45 @@ func (s *Service) GetComponentConfiguration(query *componentcfg.Query) (payload
 	return
 }
 
+func (s *Service) GetComponentConfigurationWithLastIndex(query *componentcfg.Query) (payload string, lastIndex uint64, err error) {
```

Comment:
Not urgent but perhaps `GetComponentConfiguration` above could internally call `GetComponentConfigurationWithLastIndex` to avoid code duplication.